From 19574386a45c1716898e4a9e034119937f0a70d2 Mon Sep 17 00:00:00 2001
From: Dmitriy Rabotyagov <dmitriy.rabotyagov@cleura.com>
Date: Mon, 17 Jun 2024 15:25:06 +0200
Subject: [PATCH] Allow to override Fedora subrelease regexp

This gives more flexability into defenition and won't result in users to
be blocked until new DIB release will happen with fixed regexp.

Change-Id: I5b6d15dc7cbc6f41b600c5b33f5030df5bdfa793
---
 diskimage_builder/elements/fedora/root.d/10-fedora-cloud-image | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/diskimage_builder/elements/fedora/root.d/10-fedora-cloud-image b/diskimage_builder/elements/fedora/root.d/10-fedora-cloud-image
index d401f701..5a2fb4f7 100755
--- a/diskimage_builder/elements/fedora/root.d/10-fedora-cloud-image
+++ b/diskimage_builder/elements/fedora/root.d/10-fedora-cloud-image
@@ -53,7 +53,8 @@ else
     esac
     # We have curl write headers to stderr so that we can debug fedora
     # mirror locations that don't have valid subreleases in their paths.
-    SUBRELEASE=$(head -1 < <(curl -Lis -D /dev/stderr $DIB_CLOUD_IMAGES/ | grep -o -P '(?<=Fedora-Cloud-Base-'${DIB_RELEASE}'-).*?(?=.'${ARCH}'.qcow2)' | sort -r))
+    SUBRELEASE_REGEXP=${SUBRELEASE_REGEXP:-'(?<=Fedora-Cloud-Base-'${DIB_RELEASE}'-).*?(?=.'${ARCH}'.qcow2)'}
+    SUBRELEASE=$(head -1 < <(curl -Lis -D /dev/stderr $DIB_CLOUD_IMAGES/ | grep -o -P $SUBRELEASE_REGEXP | sort -r))
     BASE_IMAGE_FILE=${BASE_IMAGE_FILE:-Fedora-Cloud-Base-$DIB_RELEASE-$SUBRELEASE.$ARCH.qcow2}
     BASE_IMAGE_TAR=Fedora-Cloud-Base-$DIB_RELEASE-$SUBRELEASE.$ARCH.tgz
     IMAGE_LOCATION=$DIB_CLOUD_IMAGES/$BASE_IMAGE_FILE
-- 
2.30.2

